core/src/main/scala/org/http4s/blaze/channel/nio1/NIO1HeadStage.scala

```diff
@@ -125,6 +102,34 @@ private[nio1] abstract class NIO1HeadStage(ch: SelectableChannel,
     }
   }
 
+  ///  channel reading bits //////////////////////////////////////////////
+
+  final override def readRequest(size: Int): Future[ByteBuffer] = {
```

Comment:
Size is not used (other than for logging. For readability it would be better to not take such parameter as now the signature of the method gives the impression that some logic is contained to attempt reading `size` bytes.

Reply:
That was the original idea of the `HeadStage.read(Int)` method, but as it turns out, it's not all that useful. I'd love to remove it, and I just filed a ticket for it: https://github.com/http4s/blaze/issues/166.